src/Components/DrillPageMinimal.js

```diff
@@ -0,0 +1,123 @@
+  };
+};
+
+export const DrillPageMinimal = props => {
+  const { route, navigation, drills } = props;
+  const { drill, training } = route.params;
+
+  const totalDrills = training.drills.length;
+  const currentDrillIndex = training.drills.findIndex(id => id === drill.id);
+
+  if (currentDrillIndex === -1) navigation.navigate('TrainingPage', { training });
+
+  const onProgressDotPress = idx => {
+    const drill = drills.find(drill => drill.id === training.drills[idx]);
+    navigation.navigate('DrillPageMinimal', { drill, training });
+  };
+
+  const goToFullDrill = () => {
+    navigation.navigate('DrillPage', { drill });
+  };
+
+  const onHeaderCheckPress = useCallback(() => {
+    if (currentDrillIndex === totalDrills - 1) {
+      navigation.navigate('TrainingPage', { training });
+    } else {
+      const nextDrill = drills.find(drill => drill.id === training.drills[currentDrillIndex + 1]);
+      navigation.navigate('DrillPageMinimal', { drill: nextDrill, training });
+    }
+  }, [training, currentDrillIndex, drills, navigation, totalDrills]);
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerRight: () => <HeaderButton onPress={onHeaderCheckPress} />,
+    });
+  }, [navigation, onHeaderCheckPress]);
+
+  return (
+    <ScrollView style={styles.drillPage}>
+      <View style={styles.titleContainer}>
+        <Text style={styles.title}>
+          Drill {currentDrillIndex + 1} - {drill.title}
+        </Text>
+        <Progress total={totalDrills} current={currentDrillIndex + 1} onDotPress={onProgressDotPress} />
+      </View>
+      <MinimalDrill drill={drill} />
+      <View style={styles.btnMoreContainer}>
+        <TouchableOpacity style={styles.btnMore} onPress={goToFullDrill}>
+          <Text style={styles.btnMoreContent}>More details on this drill</Text>
+        </TouchableOpacity>
+      </View>
+    </ScrollView>
+  );
+};
+
+export default connect(mapStateToProps)(DrillPageMinimal);
+
+const styles = StyleSheet.create({
+  drillPage: {
+    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
+    flex: 1,
+  },
+  title: {
+    fontSize: 20,
+    textAlign: 'center',
+  },
+  titleContainer: {
+    margin: 10,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  description: {
+    padding: 5,
+    paddingLeft: 20,
+    maxHeight: '20%',
+  },
+  descriptionItem: {
+    marginBottom: 5,
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  descriptionTitle: {
+    flexGrow: 0,
+    flexShrink: 0,
+    width: '25%',
+    fontWeight: 'bold',
+    color: theme.COLOR_PRIMARY,
+  },
+  descriptionText: {
+    flexGrow: 1,
+    color: theme.COLOR_SECONDARY,
+  },
+
+  btnMoreContainer: {
+    flex: 1,
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  btnMore: {
+    width: '80%',
+    backgroundColor: theme.COLOR_PRIMARY,
+    textAlign: 'center',
+    padding: 20,
+  },
+  btnMoreContent: {
+    color: 'rgb(255, 255, 255)',
```

Comment:
Maybe use a color from the theme ?

Reply:
I need to do a whole lot of styling on this view later on, but i believe we should merge as-is to have the full navigation flow and avoid conflicts on the training view.

Reply:
Yep, that was a non-blocking comment for me :)